### DISCUSSION TITLE
Set Foxglove Studio Variable from URL Parameter

### DISCUSSION THREAD

**Discussion by turtle-operator** [foxglove/community#29] [Q&A]
I would like to be able to set an arbitrary variable in Foxglove Studio to a value handed over via URL when loading the app, e.g.
`https://studio.foxglove.dev/?my_variable=42`.
I believe with the current implementation this is the same as modifying aspects of the layout via URL parameters?

**Comment by amacneil** [comment]
This is an interesting idea!

What would you expect to happen after the page loads?
- Should we just initialize the variable in Studio one time and remove the parameter from the URL?
- Or should we keep the URL parameter in sync with the variable as the user changes it?

**Reply by turtle-operator** [reply]
The parameter can be removed from the URL afterwards. It would be just important to initialize the variable in Studio one time, such that its value can be used e.g. in custom panels. We basically want to forward some meta data into Studio this way.
